src/SeqCli/Cli/Commands/Bench/BenchCommand.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2018 Datalust Pty Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Seq.Api.Model.Signals;
+using SeqCli.Cli.Features;
+using SeqCli.Connection;
+using Serilog;
+using Serilog.Context;
+using Serilog.Core;
+
+namespace SeqCli.Cli.Commands;
+
+/*
+ * Run performance benchmark tests against a Seq server.
+ *
+ * Requires test cases in a JSON file matching the format of `BenchCases.json`.
+ * 
+ * If a Seq reporting server is configured this command logs data such as:
+ * 
+ * {
+    "@t": "2022-11-09T01:12:06.0293545Z",
+    "@mt": "Bench run {BenchRunId} for query {Id}. Mean {MeanElapsed:N0} ms with relative dispersion {RelativeStandardDeviationElapsed:N2}",
+    "@m": "Bench run \"2907\" for query \"with-signal\". Mean 4 ms with relative dispersion 0.06",
+    "@i": "bb0c84a5",
+    "@r": [
+        "4",
+        "0.06"
+    ],
+    "BenchRunId": "2907",
+    "End": "2022-08-15T00:00:00.0000000",
+    "Id": "with-signal",
+    "LastResult": 606,
+    "MaxElapsed": 4.082,
+    "MeanElapsed": 3.6676,
+    "MinElapsed": 3.4334,
+    "Query": "select count(*) from stream where @Level = 'Warning'",
+    "RelativeStandardDeviationElapsed": 0.05619408341421253,
+    "Runs": 10,
+    "Signals": [
+        "signal-m33302"
+    ],
```

Comment:
We used to work on lists of intersected signals in a lot of places, but flushed them out when we moved to signal expressions (which support `or` as well as `and`). If we want to allow the use of signal names here then a structured approach would work, but otherwise, if we use ids, perhaps we just encode a signal expression in a single string value?